### PR TITLE
Add code to convert old store to zarr store

### LINKED ISSUE #964: Add code to convert object stores to zarr
Add Brendan's code for converting from one object store to the zarr format store.

### PR DESCRIPTION
**Pull request by gareth-j**
* **Summary of changes** 

This adds in some functionality to convert an old style object store to the new Zarr format. This uses some of Brendan's code for the lookup of arguments etc but I've modified it so it just takes everything in an object store and converts it.
It may be a bit temperamental and I've added some log warnings to tell people it's better to just repopulate using the original data. I've also added a short tutorial to cover how to use the new function.

* **Please check if the PR fulfills these requirements**

- [x] Closes #964
- [x] All code checks passing - `black --line-length 110` run over code, `mypy` and `flake8` not showing any errors
- [x] Documentation and tutorials updated/added
- [x] Added an entry in the latest `CHANGELOG.md` file if fixing a bug or adding a new feature

### CODE REVIEW

File: openghg/standardise/surface/_openghg.py [hunk 1]
```diff
@@ -89,6 +89,7 @@ def parse_openghg(
     # Run some checks on the
     data_attrs = {k.lower().replace(" ", "_"): v for k, v in data.attrs.items()}
 
+    print(metadata_initial)
```
Comment (brendan-m-murphy) [inline openghg/standardise/surface/_openghg.py:92]: Is this for debugging?

**Review by brendan-m-murphy — CHANGES_REQUESTED** [review]
Looks good! Just a few small suggestions for convert function.

File: openghg/store/storage/_convert.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+            data_folderpath = Path(old_store_path).joinpath("data", "uuid", uuid)
+
+            if "latest_version" in record:
+                version = record["latest_version"]
+            else:
+                try:
+                    version = max([v.name for v in data_folderpath.iterdir()], key=lambda x: int(x[1:]))
+                except (ValueError, AttributeError):
+                    # .iterdir gives empty sequence or some version name doesn't match r"v\d+" (e.g. "v10", etc.)
+                    raise ValueError("No data versions found for this search result.")
```
[the remark this answers is not in the document]
Comment (gareth-j) [reply openghg/store/storage/_convert.py:119]: Done.

File: openghg/store/storage/_convert.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+            version_folderpath = data_folderpath.joinpath(version)
+            data_filepaths = sorted(list(version_folderpath.glob("*._data")))
+
+            # Now we handle the metadata for the args we want to pass into the standardise functions
+            standardise_kwargs = {k: v for k, v in record.items() if k in valid_args}
+
+            # Parse some values to the format required for standardising
+            tf_none = {"true": True, "false": False, "none": None}
+            for k, v in standardise_kwargs.items():
+                if not isinstance(v, bool) and v.lower() in tf_none:
```
[the remark this answers is not in the document]
Comment (gareth-j) [reply openghg/store/storage/_convert.py:131]: Ah yeah good catch.

File: openghg/store/storage/_convert.py [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+                    standardise(
+                        data_type=data_type,
+                        store=store_out,
+                        filepath=data_filepaths,
+                        source_format="openghg",
+                        chunks=chunks,
+                        **standardise_kwargs,
+                    )
+                except DataOverlapError:
+                    logger.warning(f"Data overlap error for record {uuid}")
```
[the remark this answers is not in the document]
Comment (gareth-j) [reply openghg/store/storage/_convert.py:157]: Done.

File: openghg/standardise/surface/_openghg.py (hunk 1 above)
Comment (gareth-j) [reply openghg/standardise/surface/_openghg.py:92]: Oops! Removed.

File: openghg/store/storage/_convert.py [hunk 5]
```diff
@@ -116,7 +116,10 @@ def convert_store(
                     version = max([v.name for v in data_folderpath.iterdir()], key=lambda x: int(x[1:]))
                 except (ValueError, AttributeError):
                     # .iterdir gives empty sequence or some version name doesn't match r"v\d+" (e.g. "v10", etc.)
-                    raise ValueError("No data versions found for this search result.")
+                    logger.warning(
+                        f"Unable to find a data sversion for {data_type} data with UUID {uuid}, skipping..."
```
Comment (brendan-m-murphy) [inline openghg/store/storage/_convert.py:120]:
```suggestion
                        f"Unable to find a data version for {data_type} data with UUID {uuid}, skipping..."
```

**Review by brendan-m-murphy — APPROVED** [review]
Looks good! (There's a typo in an error message, but hopefully you can just commit the suggestion and merge)

**Pull request merged**